### PR TITLE
IA-3120 Allow requests that has proper Authorization header

### PR DESCRIPTION
**Pull request by Qi77Qi**
https://broadworkbench.atlassian.net/browse/IA-3120

I also reenabled spotless in this PR, so there're some pure formatting diff

### CODE REVIEW

File: service/src/main/java/org/broadinstitute/listener/relay/inspectors/TokenChecker.java [hunk 1]
```diff
@@ -47,7 +47,7 @@ private Optional<Instant> tryDecodeAsB2cToken(String token) {
       var decoded = JWT.decode(token);
       return Optional.of(decoded.getExpiresAt().toInstant());
     } catch (com.auth0.jwt.exceptions.JWTDecodeException e) {
-      logger.info("Fail to check decode JWT", e);
+      logger.debug("Fail to check decode JWT", e);
```
Comment (giventocode) [inline service/src/main/java/org/broadinstitute/listener/relay/inspectors/TokenChecker.java:50]: Shouldn't this be logger.error?

File: service/src/main/java/org/broadinstitute/listener/relay/inspectors/SamPermissionInspector.java [hunk 2]
```diff
@@ -58,12 +54,23 @@ public boolean checkCachedPermission(String accessToken) {
     return expiresAt.isAfter(Instant.now());
   }
 
-  protected Optional<String> getToken(String cookieValue) {
+  protected Optional<String> getToken(Map<String, String> headers) {
+    return getTokenFromCookie(headers).or(() -> getTokenFromAuthorization(headers));
```
Comment (giventocode) [inline service/src/main/java/org/broadinstitute/listener/relay/inspectors/SamPermissionInspector.java:58]: This means the check order would be cookie and then Auth header. I think the reverse may be better. The Auth header is the standard way of checking out. The cookie should be the fallback.  How does Leo currently do it?

File: service/src/main/java/org/broadinstitute/listener/relay/inspectors/SamResourceClient.java [hunk 3]
```diff
@@ -30,9 +30,11 @@ public Instant checkWritePermission(String accessToken) {
     try {
       var oauthInfo = tokenChecker.getOauthInfo(accessToken);
       if(oauthInfo.expiresAt().isPresent()) {
+
         samClient.setAccessToken(accessToken);
         var resourceApi = new ResourcesApi(samClient);
 
+        logger.info("token "+accessToken);
```
Comment (giventocode) [inline service/src/main/java/org/broadinstitute/listener/relay/inspectors/SamResourceClient.java:37]: Do we want to log the access token? That is sensitive information.

**Review by giventocode — CHANGES_REQUESTED** [review]

File: service/src/main/java/org/broadinstitute/listener/relay/inspectors/TokenChecker.java (hunk 1 above)
Comment (Qi77Qi) [reply service/src/main/java/org/broadinstitute/listener/relay/inspectors/TokenChecker.java:50]: I'm testing things out still...making it debug becuz I'm testing with google token..just to reduce some noise

File: service/src/main/java/org/broadinstitute/listener/relay/inspectors/SamResourceClient.java (hunk 3 above)
Comment (Qi77Qi) [reply service/src/main/java/org/broadinstitute/listener/relay/inspectors/SamResourceClient.java:37]: we don't..it's just for debugging cuz for some reason things are not working as I'm testing e2e

**Pull request merged**